### ISSUE TITLE
divide by zero error causing metric_tank crash

### ISSUE THREAD

**Issue by ctdk** [raintank/raintank-metric#90]
metric_tank in dev was running without a primary for part of today. While investigating, I found this crash in the logs:

```2015/12/31 09:00:45 http: panic serving 10.240.139.152:48781: runtime error: integer divide by zero
goroutine 508889 [running]:
net/http.(*conn).serve.func1(0xc860c03340, 0x7f67eb6f1168, 0xc83b3d9468)
        /usr/local/go/src/net/http/server.go:1287 +0xb5
main.alignRequests(0xc83984f800, 0x6, 0x6, 0xc800008598, 0xc82010c8d0, 0x3, 0x4, 0x0, 0x0, 0x0, ...)
        /home/ubuntu/.go_workspace/src/github.com/raintank/raintank-metric/metric_tank/query_engine.go:147 +0x79c
main.Get(0x7f67e88d36f0, 0xc860c03550, 0xc83606f6c0, 0xc82047bba0, 0xc82010c8d0, 0x3, 0x4)
        /home/ubuntu/.go_workspace/src/github.com/raintank/raintank-metric/metric_tank/http.go:140 +0x11cf
main.get.func1(0x7f67e88d36f0, 0xc860c03550, 0xc83606f6c0)
        /home/ubuntu/.go_workspace/src/github.com/raintank/raintank-metric/metric_tank/http.go:36 +0x60
net/http.HandlerFunc.ServeHTTP(0xc820c171a0, 0x7f67e88d36f0, 0xc860c03550, 0xc83606f6c0)
        /usr/local/go/src/net/http/server.go:1422 +0x3a
net/http.(*ServeMux).ServeHTTP(0xc820051dd0, 0x7f67e88d36f0, 0xc860c03550, 0xc83606f6c0)
        /usr/local/go/src/net/http/server.go:1699 +0x17d
net/http.serverHandler.ServeHTTP(0xc820013e00, 0x7f67e88d36f0, 0xc860c03550, 0xc83606f6c0)
        /usr/local/go/src/net/http/server.go:1862 +0x19e
net/http.(*conn).serve(0xc860c03340)
        /usr/local/go/src/net/http/server.go:1361 +0xbee
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:1910 +0x3f6
2015/12/31 09:00:45 ^[[1;32m[I] terminating.^[[0m```

This sort of crash shows up numerous times in the log, all involving `raintank-metric/metric_tank/query_engine.go:147`.

**Comment by Dieterbe** [comment]
I think this bug was introduced in 45f40182b36500cafdfac1058aeaeec9f57d000a, 18 hours ago, and fixed by 34f7efc0dda330619d7bd11e06f8e31534eca80b, 16 hours ago.
a narrow time margin, but is it possible that the former was deployed, and wasn't updated once the bugfix was committed?

**Comment by ctdk** [comment]
The last time it happened was at 9:00am UTC on metric-tank-1 and 8:20am UTC on metric-tank-2, so I think that that did in fact fix it (although the timeframe isn't quite right, I also recall that chef should never restart metric-tank, so the fix didn't show up until metric-tank had crashed, and it restarted with the fixed binary).

I'll check in tomorrow to see if any more of these crashes have happened, and if they haven't I'll close this out. I think the problem's already been fixed.
